common/djangoapps/entitlements/models.py

```diff
@@ -320,6 +329,99 @@ def get_active_entitlements_for_user(cls, user):
             enrollment_course_run=None
         ).select_related('user').select_related('enrollment_course_run')
 
+    @classmethod
+    def get_fulfillable_entitlements(cls, user):
+        """
+        Returns all fulfillable entitlements for a User
+
+        Arguments:
+            user (User): The user we are looking at the entitlements of.
+
+        Returns
+            list<CourseEntitlements>: A list of course Entitlements that a user can enroll in. These must not be expired
+                and not have a course run already assigned to it.
+        """
+
+        try:
+            return cls.objects.filter(
+                user=user,
+            ).exclude(
+                expired_at__isnull=False,
+                enrollment_course_run__isnull=False
+            ).order_by('created')
+        except CourseEntitlement.DoesNotExist:
```

Comment:
Correct me if I'm wrong, but I believe that if no objects match the parameters requested when you perform filter and exclude that an empty query set will be returned without raising the DoesNotExist exception, which would make this try..except block unnecessary

Reply:
additionally, make sure `order_by('created')` is returning most recently created first,; if not, simply add a minus sign in front of the word created `order_by('-created')`